paasta_tools/cli/cmds/spark_run.py

```diff
@@ -56,6 +56,12 @@
 CLUSTER_MANAGER_MESOS = "mesos"
 CLUSTER_MANAGER_K8S = "kubernetes"
 CLUSTER_MANAGERS = {CLUSTER_MANAGER_MESOS, CLUSTER_MANAGER_K8S}
+# Reference: https://spark.apache.org/docs/latest/configuration.html#application-properties
+DEFAULT_DRIVER_CORES_BY_SPARK = 1
+DEFAULT_DRIVER_MEMORY_BY_SPARK = "1g"
+# Extra room for memory overhead and for any other running inside containe
+DOCKER_RESOURCE_ADJUSTMENT_FACTOR = 2
```

Comment:
There is driver.memory.overheadfactor which also uses driver memory (default 10%) and some users can have higher value. Accounting that into the calculation can make the code complicated.

Question: If the docker container memory limit is set to 4G (let's say), and the box has 40G storage. Does that mean only 10 docker containers will be allocated at max? Or 20 docker containers could be allocated too, but none of them would exceed 4G.

Reply:
I make it configurable optionally.

Reply:
for your question: since we're running these containers directly, it'd be the latter - there's nothing stopping Tron's SSH code from starting more docker containers (with a 4GB memory limit) than would actually fit onto a box. 

(e.g., it would happily try to start 100 containers with a memory limit of 4GB)

Setting this limit won't solve all the problems with running the drivers on bare-metal because of this, but it should significantly improve things when one driver starts hogging all the memory

Reply:
Makes sense. That's what I thought too. Thanks for confirming :)

Reply:
I was thinking of making this configurable through Puppet (using https://github.com/Yelp/paasta/blob/master/paasta_tools/utils.py#L1847) so that we don't need to release new paasta versions to tweak this factor - but this is fine for now